rdrf/rdrf/services/io/defs/importer.py

```diff
@@ -351,29 +352,35 @@ def _create_cdes(self, cde_maps):
                         registries_already_using)
 
             for field in cde_map:
-                if field not in ["code", "pv_group"]:
-                    import_value = cde_map[field]
-                    if not created:
-                        old_value = getattr(cde_model, field)
-                        if old_value != import_value:
-                            logger.warning(
-                                "import will change cde %s: import value = %s new value = %s" %
-                                (cde_model.code, old_value, import_value))
-
-                    if field == 'datatype':
-                        import_value = CdeMappings.fix_data_type(import_value.strip())
-                        valid_types = [choice[0] for choice in CommonDataElement.DATA_TYPE_CHOICES]
-                        if import_value not in valid_types:
-                            raise ValidationError(f'Invalid data type {import_value} for CDE: {cde_map["code"]}')
-                    elif field == 'widget_name':
-                        import_value = CdeMappings.fix_widget_name(import_value.strip())
-                        data_type = CdeMappings.fix_data_type(cde_map.get('datatype', ''))
-                        valid_widgets = get_widgets_for_data_type(data_type) + ['']
-                        if import_value not in valid_widgets:
-                            raise ValidationError(f'Invalid widget_name {cde_map[field]} for datatype {data_type} and CDE: {cde_map["code"]}')
-
-                    setattr(cde_model, field, import_value)
-                    # logger.info("cde %s.%s set to [%s]" % (cde_model.code, field, cde_map[field]))
+                if not hasattr(cde_model, field):
+                    if field not in unknown_attributes:
+                        unknown_attributes.add(field)
+                        logger.warning(f"CDE model does not have the `{field}` attribute !")
+                    continue
+                if field in ["code", "pv_group"]:
+                    continue
+                import_value = cde_map[field]
+                if not created:
```

Comment:
This `if` should be at line `381` as the following "data fixes" could change the `import_value` variable's value.

Reply:
Updated.